mangdangroboticsclub/mini_pupper_2_bsp
# Fix for audio hardware driver

## Proposed change(s)

This PR updates the `rpi_i2s_audio` driver, introduces a new `prepare_dkms.py` script for easy installation and management of the driver using DKMS, and modifies the testing procedure.

### Useful links (GitHub issues, forum threads, etc.)

_None provided_

### Types of change(s)


## Testing and Verification

Please follow the instructions below to test and verify the changes.

1. Run `prepare_dkms.py` script
2. Run `install.py` script
3. Reboot
4. Run `audio_test.sh` script from the `demo` folder
5. Check if the `rpi_i2s_audio` driver is installed, loaded correctly, and the audio test is successful

## Test Configuration

__Mini Pupper Version__  
Mini Pupper 2

__Raspberry Pi OS + ROS version__  
Ubuntu 22.04, ROS 2 Humble

__PC OS + ROS version__  
Ubuntu 22.04, ROS 2 Humble

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/mangdangroboticsclub/mini_pupper_2_bsp/blob/main/CONTRIBUTING.md) guidelines.

## Other comments

This PR includes changes to the `dkms.conf` file, the new `prepare_dkms.py` script that automates the installation and management of the `rpi_i2s_audio` driver using DKMS, and updates the testing procedure to include running `install.py` script and `audio_test.sh` script from the `demo` folder.
